### DISCUSSION TITLE
Module 'path_provider_foundation' not found

### DISCUSSION THREAD

**Discussion by 33885222** [codemagic-ci-cd/codemagic-docs#2609] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

The system cannot build the iOS file from the Flutter project.
It throws the error: Parse Issue (Xcode): Module 'path_provider_foundation' not found.
Even though the dependencies are specified in the pubspec.yaml file:

dependencies:
  flutter:
    sdk: flutter

  path_provider: ^2.1.3
  path_provider_foundation: ^2.4.0

What could be the problem?

### Expected results

Successful build

### Actual results

Installing dependencies
13s

== Install Flutter dependencies ==
> flutter packages pub get
Resolving dependencies...
Downloading packages...
  collection 1.18.0 (1.19.0 available)
  leak_tracker 10.0.4 (10.0.5 available)
  leak_tracker_flutter_testing 3.0.3 (3.0.5 available)
  material_color_utilities 0.8.0 (0.12.0 available)
  meta 1.12.0 (1.15.0 available)
  test_api 0.7.0 (0.7.2 available)
  vm_service 14.2.1 (14.2.4 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.

Building iOS
23s

== Building for iOS ==

> flutter build ios --debug --no-codesign
Resolving dependencies...
Downloading packages...
  collection 1.18.0 (1.19.0 available)
  leak_tracker 10.0.4 (10.0.5 available)
  leak_tracker_flutter_testing 3.0.3 (3.0.5 available)
  material_color_utilities 0.8.0 (0.12.0 available)
  meta 1.12.0 (1.15.0 available)
  test_api 0.7.0 (0.7.2 available)
  vm_service 14.2.1 (14.2.4 available)
Got dependencies!
7 packages have newer versions incompatible with dependency constraints.
Try `flutter pub outdated` for more information.
Warning: Building for device with codesigning disabled. You will have to manually codesign before deploying to device.
Building com.example.totoVpn for device (ios)...
Running pod install...                                             557ms
Running Xcode build...                                          
Xcode build done.                                           17.1s
Failed to build iOS app
Parse Issue (Xcode): Module 'path_provider_foundation' not found
/Users/builder/clone/ios/Runner/GeneratedPluginRegistrant.m:11:8

Encountered error while building for device.

Build failed :|
Failed to build for iOS

### Build id (optional)

667abfb82193defe8d33998c

**Comment by 33885222** [comment, marked as answer]
It looks like I found a similar issue: https://github.com/orgs/codemagic-ci-cd/discussions/2053
This solution helped me, however, my dependencies require a newer version of iOS, but by default it installs version 12. So, I had to amend the script as follows:
```

rm -rf ios
flutter create /Users/builder/clone/
podfile_path="/Users/builder/clone/ios/Podfile"

if [ -f "$podfile_path" ]; then
  echo "Podfile"
  platform_line_exists=$(grep -q "platform :ios" "$podfile_path")
  if grep -q "platform :ios" "$podfile_path"; then
    echo "'platform :ios' HERE"
    sed -i '' '/platform :ios/d' "$podfile_path"
  else
    echo "platform :ios' NO"
  fi
  
  sed -i '' '1s/^/platform :ios, '"'"'15.0'"'"'\n/' "$podfile_path"
  
else
  echo "no Podfile"

  touch "$podfile_path"
  echo "platform :ios, '15.0'" >> "$podfile_path"
fi

echo "## Podfile:"
cat "$podfile_path"
```

**Reply by himesh-cm** [reply]
Great, Thanks for the brief response.
While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!
